[BUG] rapids-tools v22.04.0 release jar reports maven dependency issue :  rapids-4-spark-common_2.12:jar:22.04.0 NOT FOUND

rapids-tools v22.04.0 release jar reports maven dependency issue :  rapids-4-spark-common_2.12:jar:22.04.0 `NOT FOUND`

CLI: 
```
mvn dependency:get -B -DremoteRepositories=https://oss.sonatype.org/content/repositories/staging -DgroupId=com.nvidia -DartifactId=rapids-4-spark-tools_2.12 -Dversion=22.04.0
```

Error log:
```
09:58:55  [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.8:get (default-cli) on project standalone-pom: Couldn't download artifact: Missing:
09:58:55  [ERROR] ----------
09:58:55  [ERROR] 1) com.nvidia:rapids-4-spark-common_2.12:jar:22.04.0
09:58:55  [ERROR] 
09:58:55  [ERROR] Try downloading the file manually from the project website.
09:58:55  [ERROR] 
09:58:55  [ERROR] Then, install it using the command:
09:58:55  [ERROR] mvn install:install-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-common_2.12 -Dversion=22.04.0 -Dpackaging=jar -Dfile=/path/to/file
09:58:55  [ERROR] 
09:58:55  [ERROR] Alternatively, if you host your own repository you can deploy the file there:
09:58:55  [ERROR] mvn deploy:deploy-file -DgroupId=com.nvidia -DartifactId=rapids-4-spark-common_2.12 -Dversion=22.04.0 -Dpackaging=jar -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
09:58:55  [ERROR] 
09:58:55  [ERROR] Path to dependency:
09:58:55  [ERROR] 1) org.apache.maven.plugins:maven-downloader-plugin:jar:1.0
09:58:55  [ERROR] 2) com.nvidia:rapids-4-spark-tools_2.12:jar:22.04.0
09:58:55  [ERROR] 3) com.nvidia:rapids-4-spark-common_2.12:jar:22.04.0
09:58:55  [ERROR] 
09:58:55  [ERROR] ----------
09:58:55  [ERROR] 1 required artifact is missing.
```

Root cause: 
rapids-tools:v22.04.0 release candidates depends on `rapids-4-spark-common_2.12:jar:22.04.0`, but we won't release apids-4-spark-common_2.12:v22.04.0 to maven central: 

https://github.com/NVIDIA/spark-rapids/blob/branch-22.04/tools/pom.xml#L45 
```
        <dependency>
            <groupId>com.nvidia</groupId>
            <artifactId>rapids-4-spark-common_${scala.binary.version}</artifactId>
            <version>${project.version}</version>
        </dependency>
```

--- a/tools/pom.xml
+++ b/tools/pom.xml
@@ -41,11 +41,6 @@
 
     <dependencies>
         <dependency>
-            <groupId>com.nvidia</groupId>
-            <artifactId>rapids-4-spark-common_${scala.binary.version}</artifactId>
-            <version>${project.version}</version>
-        </dependency>
-        <dependency>
             <groupId>org.scala-lang</groupId>
             <artifactId>scala-library</artifactId>
             <scope>compile</scope>
@@ -105,7 +100,6 @@
                             <artifactSet>
                                 <includes>
                                     <include>org.rogach:scallop_${scala.binary.version}</include>
-                                    <include>com.nvidia:rapids-4-spark-common_${scala.binary.version}</include>
                                 </includes>
                             </artifactSet>
                             <filters>

--- a/tools/src/main/scala/com/nvidia/spark/rapids/tool/ThreadFactoryBuilder.scala
+++ b/tools/src/main/scala/com/nvidia/spark/rapids/tool/ThreadFactoryBuilder.scala
@@ -1,0 +1,52 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tool
+
+import java.util.concurrent.{Executors, ThreadFactory}
+import java.util.concurrent.atomic.AtomicLong
+
+// This is similar to Guava ThreadFactoryBuilder
+// Avoid to use Guava as it is a messy dependency in practice.
+// This is copied from the common module
+class ThreadFactoryBuilder {
+  private var nameFormat = Option.empty[String]
+  private var daemon = Option.empty[Boolean]
+
+  def setNameFormat(nameFormat: String): ThreadFactoryBuilder = {
+    nameFormat.format(0)
+    this.nameFormat = Some(nameFormat)
+    this
+  }
+
+  def setDaemon(daemon: Boolean): ThreadFactoryBuilder = {
+    this.daemon = Some(daemon)
+    this
+  }
+
+  def build(): ThreadFactory = {
+    val count = nameFormat.map(_ => new AtomicLong(0))
+    new ThreadFactory() {
+      private val defaultThreadFactory = Executors.defaultThreadFactory
+
+      override def newThread(r: Runnable): Thread = {
+        val thread = defaultThreadFactory.newThread(r)
+        nameFormat.foreach(f => thread.setName(f.format(count.get.getAndIncrement())))
+        daemon.foreach(b => thread.setDaemon(b))
+        thread
+      }
+    }
+  }
+}

--- a/tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala
+++ b/tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/Profiler.scala
@@ -22,8 +22,8 @@ import scala.collection.JavaConverters._
 import scala.collection.mutable.{ArrayBuffer, HashMap}
 import scala.util.control.NonFatal
 
-import com.nvidia.spark.rapids.ThreadFactoryBuilder
 import com.nvidia.spark.rapids.tool.{EventLogInfo, EventLogPathProcessor}
+import com.nvidia.spark.rapids.tool.ThreadFactoryBuilder
 import org.apache.hadoop.conf.Configuration
 
 import org.apache.spark.internal.Logging

--- a/tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala
+++ b/tools/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala
@@ -20,8 +20,8 @@ import java.util.concurrent.{ConcurrentLinkedQueue, Executors, ThreadPoolExecuto
 
 import scala.collection.JavaConverters._
 
-import com.nvidia.spark.rapids.ThreadFactoryBuilder
 import com.nvidia.spark.rapids.tool.EventLogInfo
+import com.nvidia.spark.rapids.tool.ThreadFactoryBuilder
 import org.apache.hadoop.conf.Configuration
 
 import org.apache.spark.internal.Logging

--- a/tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala
+++ b/tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala
@@ -22,8 +22,8 @@ import java.util.regex.PatternSyntaxException
 
 import scala.collection.JavaConverters._
 
-import com.nvidia.spark.rapids.ThreadFactoryBuilder
 import com.nvidia.spark.rapids.tool.EventLogInfo
+import com.nvidia.spark.rapids.tool.ThreadFactoryBuilder
 import com.nvidia.spark.rapids.tool.qualification.QualificationArgs
 import org.apache.hadoop.conf.Configuration
 

